### PR TITLE
git-extras: Add ZSH completions

### PR DESCRIPTION
**Pull request by tresni**
ZSH completions are available in the 4.4.0 release.  This simply installs them correctly.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@zmwangx should we tweak the file name on install?

**Comment by zmwangx** [comment]
`git-extras-completion.zsh` isn't the usual, autoloadable type of zsh completion definition you see elsewhere. To make things work automatically, all of `_git-$subcommand` functions needs to be in separate files, which isn't happening; so `git-extras-completion.zsh` needs to be sourced manually. Putting it in `site-functions` would be misleading because it's not a function.

What I would do is

```rb
    pkgshare.install "etc/git-extras-completion.zsh"
```

with a caveat

```rb
  def caveats; <<-EOS.undent
    To load Zsh completions, add the following to your .zshrc:
      . #{HOMEBREW_PREFIX}/share/git-extras/git-extras-completion.zsh
    EOS
  end
```

By the way, this formula's `install` puts a bunch of shell scripts and man pages in place without any compilation, so I would switch to `bottle :unneeded` too.

**Comment by tresni** [comment]
I am using this as I have suggested in the patch.  It does not require additional sourcing of the script in zsh as long as /usr/local/share/zsh/site-functions is in `$fpath`.  That's why I did it this way instead of putting it in pkgshare with a message (which was my initial idea.)  Not sure how I prove that this works (asciinema?)

**Comment by zmwangx** [comment]
> It does not require additional sourcing of the script in zsh as long as `/usr/local/share/zsh/site-functions` is in `$fpath`.

Well, Zsh does not automatically source everything in `$fpath`, or mark every function in every file as autoload... There are basically two ways to define a function: one, just define it, so you need to source the definition at some point; two, autoload it, the rule of which is documented very clearly [here](http://zsh.sourceforge.net/Doc/Release/Functions.html), and having a single file in fpath that defines a load of functions doesn't really work (unless it's a compiled `.zwc` file, which is another story).

The fact that you're able to complete `git extras`, i.e., you're able to load `_git-extras` somehow, is curious. You can find out where it's defined and when it's loaded with `type _git-extras`. Basically, start out fresh, do the following:

```console
$ type _git-extras  # You should see the following if _git-extras is properly marked as autoload; if you see an actual path, then that path has already been sourced.
_git is an autoload shell function
$ git ext<tab>
$ type _git-extras  # This time you should see the path, if the function is properly autoloaded.
_git-extras is a shell function from /usr/local/share/zsh/site-functions/git-extras-completion.zsh
```

Also, do you use any Zsh configuration framework? It could be doing something for you.

**Comment by DomT4** [comment]
> . #{HOMEBREW_PREFIX}/share/git-extras/git-extras-completion.zsh

```ruby
. #{opt_pkgshare}/git-extras-completion.zsh
```
?

**Comment by zmwangx** [comment]
> `#{opt_pkgshare}/git-extras-completion.zsh`

Huh, didn't even know `opt_pkgshare` is a thing. I blame <https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#variables-for-directory-locations>.

That said, `/usr/local/share/zsh/site-functions/git-extras-completion.zsh` looks so much nicer than `/usr/local/opt/git-extras/share/git-extras/git-extras-completion.zsh`.

**Comment by DomT4** [comment]
> Huh, didn't even know opt_pkgshare is a thing. I blame https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#variables-for-directory-locations.

Yeah that's pretty incomplete. `opt_include`, `opt_lib`, `opt_libexec`, `opt_pkgshare`, `opt_share`, etc all work.

**Comment by tresni** [comment]
@zmwangx did some digging and you are right.  This isn't doing what I expected.  I will adjust

**Review by ilovezfs — CHANGES_REQUESTED** [review]
It actually can't be switched to bottle unneeded unless we can figure out how to do the installation without using `make`

**Comment by tresni** [comment]
@ilovezfs reverted `bottle :unneeded` and fixed `audit --strict` message about caveat coming after test.

File: Formula/git-extras.rb [hunk 1]
```diff
@@ -18,13 +18,21 @@ class GitExtras < Formula
     sha256 "59e7af5b051fcab03f394ebb8627d5f41f9b9c3be543b59329c9356e1565eee8" => :sierra
     sha256 "59e7af5b051fcab03f394ebb8627d5f41f9b9c3be543b59329c9356e1565eee8" => :el_capitan
     sha256 "59e7af5b051fcab03f394ebb8627d5f41f9b9c3be543b59329c9356e1565eee8" => :yosemite
+    rebuild 1
```
Comment (DomT4) [inline Formula/git-extras.rb]: Don't need this line added.

Comment (tresni) [reply Formula/git-extras.rb]: Version is not changing so without this line it doesn't rebuild the bottle and will fetch the existing bottle for this specific version.  In the old days I would have marked this `revision 1` but `audit` told me to use `rebuild 1` instead.

Comment (tresni) [reply Formula/git-extras.rb]: :+1: taken care of, thanks for the info!

**Comment by tresni** [comment]
I believe I have addressed all feedback does this need anything else before it can be merged?

**Comment by DomT4** [comment]
Will need tweaking to use the new `EOS` style now. You want `<<~EOS` rather than `<<-EOS.undent` essentially.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@tresni yes this needs the need heredoc style as @DomT4 said. Also, why aren't we just using `zsh_completion.install` and a name starting with `_`?

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks @tresni! Shipped.

**Pull request closed without merging**